packages/design-system/src/components/tree/tree-node.tsx

```diff
@@ -111,30 +126,195 @@ const hoverStyle = {
   boxSizing: "border-box",
 };
 
-const ItemWrapper = styled(Flex, {
+const suffixWidthVar = cssVars.define("suffix-width");
+
+const ItemContainer = styled(Flex, {
   color: "$hiContrast",
   alignItems: "center",
   position: "relative",
+
   variants: {
     isSelected: {
       true: { color: "$loContrast", bc: "$blue10" },
     },
     parentIsSelected: {
       true: { bc: "$blue4" },
     },
+    suffixVisible: {
+      true: {
+        [`& ${ItemButton}`]: {
+          // We have to use a padding to make space for the suffix
+          // because we can't put it inside ItemButton (see comment below)
+          paddingRight: cssVars.use(suffixWidthVar, "0"),
+        },
+        [`& ${SuffixContainer}`]: {
+          opacity: "1",
+          pointerEvents: "all",
+        },
+      },
+    },
     enableHoverState: {
-      true: { "&:hover:after": hoverStyle },
+      true: {
+        "&:hover:after": hoverStyle,
+
+        // Suffix is also visible on hover, which we don't track using JavaScript
+        [`&:hover ${ItemButton}`]: {
```

Comment:
instead of the ItemButton selector we could use the CSS variables all the way, similar to how I did it with `imageInfoTriggerCssVars()`
1. component exports a high-level function that returns all variables
2. consumer uses it to set the variables

in this case it would look like this

`'&:hover': itemButtonCssVars('hover')`

below it would be:

`&:hover: suffixContainerCssVars('hover')`

Reply:
But why not use `ItemButton` selector? This is a local component, so there's no breaking of encapsulation.

Reply:
1. Because this way you can't create a different Button component that would be passed to the ItemContainer. You are coupling these.
2. We might need to get rid of stitches long term, so the less features from stitches we use the better. It will be much easier to get rid of it.
3. I think we should leverage CSS variables to the maximum, they give us both: decoupling, all native CSS things like :hover, customization and even dynamic updates. I think long term CSS props should be heavily used